tests/components/smart_meter_b_route/conftest.py

```diff
@@ -0,0 +1,46 @@
+"""Common fixtures for the Smart Meter B-route tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.smart_meter_b_route.async_setup_entry",
+        return_value=True,
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_momonga(exception=None) -> Generator[Mock]:
+    """Mock for Momonga class."""
+
+    class MockMomonga:
+        def __init__(self, *args, **kwargs) -> None:
+            if exception:
+                raise exception
+
+        def open(self):
+            pass
+
+        def get_instantaneous_current(self) -> dict[str, float]:
+            return {
+                "r phase current": 1,
+                "t phase current": 2,
+            }
+
+        def get_instantaneous_power(self) -> float:
+            return 3
+
+        def get_measured_cumulative_energy(self) -> float:
+            return 4
```

Comment:
I rewrote the tests to avoid the usage of `patch()` and have a more concise syntax.

From what I understood, `Mock` is better to mock synchronous code and `AsyncMock` for asynchronous code because it ensures it can await all the methods. Momonga being 100% synchronous, I left it as a `Mock` but I changed the way I used it.

Is the current syntax closer to what you were thinking ?

Reply:
iirc
```py
 patch(
            "homeassistant.components.smart_meter_b_route.coordinator.Momonga",
            autospec=True,
        ) as mock_momonga,
```
the autospec makes sure it picks the right mock

Reply:
This looks a lot better now indeed